### PR TITLE
Use SERVER_NAME and PREFERRED_URL_SCHEME in the composition

### PR DESCRIPTION
**Pull request by karenc**
- Remove `BASE_URL` and `MAIL_BASE_URL` and change code to use
  `flask.url_for` to generate urls
- Rename the nginx service from www to localhost
- Add port forwarding 80 to the nginx service so now the site is
  accessible at http://localhost/
- Change all the `localhost` to `127.0.0.1` in docker-compose
  healthchecks, as `localhost` is now the name of the nginx service

This attempts to use the `SERVER_NAME` and `PREFERRED_URL_SCHEME`
environment variables. This is a move towards deprecating the
`HOUSTON_URL` environment variable that sets the `BASE_URL`
configuration setting. See also
[DEX-695 Obsolete the BASE_URL configuration
setting](https://wildme.atlassian.net/browse/DEX-695)

One downside of this change is that `SERVER_NAME` setting has special
behavior within Werkzeug. The server, much like nginx or any other
server, will only serve for the given name; so the `Host:
$SERVER_NAME` request header must be supplied when making requests
within the houston container if trying to directly contact the service
without the reverse proxy.

On the upside, because we tell Werkzeug to serve under this name,
we must also be able to actual server under the name and share it with
other services (i.e. ACM in the callback url). To do this I've worked
around the issue by utilizing the name `localhost` for both internal
(to the composition's network) and external usage. For example,
requesting `http://localhost/api/v1/users/me` will work both inside the
container and outside of it.

Using this change will mean the composition is serving very similar to
how it would be served in production (or at least how Flask likes to
be served in production). And thus no workaround, other than the
service named `localhost` added in this change, will need to be
present. That means cleaner code with less special snowflakes! Yay!

### CODE REVIEW

File: app/extensions/elasticsearch/__init__.py [hunk 1]
```diff
@@ -1312,7 +1312,7 @@ def es_status(app=None, outdated=True, missing=False, active=True, health=True):
             status['elasticsearch:enabled'] = False
 
         health_data, stats_data = es_health_and_stats(app=app)
-        if health_data['status'] != 'green':
+        if health_data['status'] not in ('green', 'yellow'):
```
Comment (bluemellophone) [inline app/extensions/elasticsearch/__init__.py:1315]: I want to keep the yellow status check, but instead make the tests ignore a yellow status.  See here for potential fix: 0a9a867de6f31ba8498b0d28b3d300b77aea8e5d

Comment (bluemellophone) [reply app/extensions/elasticsearch/__init__.py:1315]: This fix has been merged into main, so this health status check can be deleted.

Comment (karenc) [reply app/extensions/elasticsearch/__init__.py]: Thanks!  I've removed the elasticsearch commits

**Review by colinwkingen — APPROVED** [review]
LGTM

**Pull request merged**